src/proxy/http/HttpSM.cc

```diff
@@ -1823,6 +1823,9 @@ HttpSM::state_http_server_open(int event, void *data)
       _netvc->do_io_write(this, 1, _netvc_reader);
       _netvc->set_inactivity_timeout(this->get_server_connect_timeout());
 
+      // Pre-emptively set a server connect failure that will be cleared once a WRITE_READY is received from origin or
+      // bytes are received back
+      t_state.set_connect_fail(EIO);
```

Comment:
I guess you're just moving existing code. However, can we set `EIO` only if we observe errors? IMO, we easily make a mistake with current approach.